File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/breakr/Break.java
package tasks.breakr;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Int32;

@JavaTaskInfo
public class Break extends JavaTask {

	@InputPortInfo(name = "In", dataType = Int32.class)
	public InputPort in;

	@Override
	public void execute() {
		int i = in.getData(Int32.class).getValue();
		if(i == 3) super.exitRunner();
	}
}

File: com.systemincloud.examples.tasks.rtask/src/test/r/break/Break.R

library(R6)
library(sicrtask)

Break <- R6Class("Break",
  inherit = RTask,
  public = list(
    rtaskinfo <- RTaskInfo$new(),
    in1 = NA,
    initialize_ports = function() {
      self$in1 <- InputPort$new(name = "in1", data_type = Int32)
    },
    initialize = function() {
      self$initialize_ports()
    },
    execute = function(grp) {
	  i = self$in1$get_data(Int32)$values[0]
	  if( i == 3 ) self$exit_runner()
	},
  ),
  private = list(
  )
)
